sympy/series/approximants.py

```diff
@@ -101,3 +108,283 @@ def approximants(l, X=Symbol('x'), simplify=False):
+    """
+    Returns all pade approximants of the expression `f` of the desired order.
+
+    Description
+    ===========
+
+    Computes all pade approximants `f` of order `[m/n]` with `m + n =` order.
+
+    Parameters
+    ==========
+
+    f : Poly
+        The polynomial to approximate, must be univariate
+    order : int
+        The desired order of the pade approximants
+
+    Returns
+    =======
+
+    pade approximants : Generator[tuple[Poly, Poly], None, None]
+        Generator for (numerator, denominator) pairs of polynomials
+        representing the numerator and denominators of the approximants.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x
+    >>> from sympy import Poly
+    >>> from sympy.series.approximants import pade_approximants_gcdex
+
+    >>> exp_taylor_poly = Poly(1 + x + x**2/2 + x**3/6 + x**4/24, x, domain='QQ')
+    >>> pade_exp = pade_approximants_gcdex(exp_taylor_poly, 2)
+    >>> for num, denom in pade_exp: print(num, denom)
+    Poly(1/2*x**2 + x + 1, x, domain='QQ') Poly(1, x, domain='QQ')
+    Poly(2*x + 4, x, domain='QQ') Poly(-2*x + 4, x, domain='QQ')
+    Poly(1, x, domain='QQ') Poly(1/2*x**2 - x + 1, x, domain='QQ')
+
+    To get pade approximants, divide the numerators by the denominators
+
+    >>> sin_taylor_poly = Poly(x - x**3/6 + x**5/120, domain='QQ')
+    >>> pade_sin = pade_approximants_gcdex(sin_taylor_poly, 5)
+    >>> for num, denom in pade_sin: print(num/denom)
+    x**5/120 - x**3/6 + x
+    -(20*x**4 - 120*x**2)/(120*x)
+    (-7*x**3/60 + x)/(x**2/20 + 1)
+    360*x**2/(7*(60*x**3/7 + 360*x/7))
+    x/(7*x**4/360 + x**2/6 + 1)
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant_gcdex
+    sympy.series.approximants.pade_approximant
+    """
+
+    if order < 0:
+        raise ValueError("'order' must be a non-negative integer")
+
+    if not f.is_univariate:
+        raise ValueError("f must be a univariate polynomial")
+
+    # truncate f so f.degree() <= order
+    if f.degree() > order:
+        f = f.slice(0, order+1)
+
+    x = f.gens[0]
+
+    remainder_monomial = Poly(x ** (order + 1), x)
+
+    yield f, f.one
+
+    euclidean_algorithm_steps = gcdex_steps(remainder_monomial, f)
+
+    for _, t, r in euclidean_algorithm_steps:
+        yield r, t
+
+
+@public
+def pade_approximant_gcdex(
+    f: Poly, m: int, n: int | None = None
+    ) -> tuple[Poly, Poly] | tuple[None, None]:
+    """
+    `[m/n]` pade approximant of `f` around `x=0`. Computed using the extended euclidean algorithm.
+
+    Description
+    ===========
+
+    For a polynomial `f` and integers `m` and `n`, the `[m/n]` pade approximant
+    of `f` is a rational function `p(x)/q(x)`, where `p(x)` and `q(x)` are
+    polynomials of degree at most `m` and `n` respectively. The pade
+    approximation, when it exists, is such that the taylor series of `p(x)/q(x)`
+    around `x=0` matches `f` up to order `(m + n)` in `x`.
```

Comment:
One of the changes I made when pushing a few commits here was to wrap long lines at 78 characters (https://github.com/sympy/sympy/pull/27671/commits/933b17f772f12f2abde38291d5287ce70807f260). This is particularly important to do in docstrings where there is no need to have such long lines and where the prose text is often displayed as tool tips in editors and things.

The first line here should be shortened rather than wrapped though.